attest/verifier/types/src/convert/enclave_report_data_contents.rs

```diff
@@ -51,11 +51,32 @@ impl From<&EnclaveReportDataContents> for prost::EnclaveReportDataContents {
         }
     }
 }
+use mc_crypto_digestible::{DigestTranscript, Digestible};
+
+impl Digestible for prost::EnclaveReportDataContents {
+    fn append_to_transcript<DT: DigestTranscript>(
+        &self,
+        context: &'static [u8],
+        transcript: &mut DT,
+    ) {
+        let typename = b"EnclaveReportDataContents";
+        transcript.append_agg_header(context, typename);
+        transcript.append_primitive(context, b"nonce", &self.nonce);
+        transcript.append_primitive(context, b"key", &self.key);
+        // Since custom identity is optional we only include it if it has data.
+        if !self.custom_identity.is_empty() {
+            transcript.append_primitive(context, b"custom_identity", &self.custom_identity);
+        }
+        transcript.append_agg_closer(context, typename);
```

Comment:
I was wondering if there's a way to protect us from forgetting to add new fields here if we ever evolve the protobuf, and came up with this with seems to work and will complain if a new field got added.

```suggestion
        let typename = b"EnclaveReportDataContents";
        transcript.append_agg_header(context, typename);

        let Self {nonce, key, custom_identity} = self;
        transcript.append_primitive(context, b"nonce", &nonce);
        transcript.append_primitive(context, b"key", &key);
        // Since custom identity is optional we only include it if it has data.
        if !self.custom_identity.is_empty() {
            transcript.append_primitive(context, b"custom_identity", &custom_identity);
        }

        transcript.append_agg_closer(context, typename);

```

Reply:
oh, that's a cool use of destructuring.  and would definitly help ensure it, especially for the upcoming `Collateral` with 7 fields that would be good to know we got them all.